cmd/payload-testing/prow-plugin/server.go

```diff
@@ -0,0 +1,85 @@
+package main
+
+import (
+	"fmt"
+	"regexp"
+
+	"github.com/openshift/ci-tools/pkg/api"
+	"github.com/openshift/ci-tools/pkg/promotion"
+	configresolver "github.com/openshift/ci-tools/pkg/registry/server"
+	"github.com/sirupsen/logrus"
+
+	prowconfig "k8s.io/test-infra/prow/config"
+	"k8s.io/test-infra/prow/github"
+	"k8s.io/test-infra/prow/pluginhelp"
+)
+
+type githubClient interface {
+	CreateComment(owner, repo string, number int, comment string) error
+}
+
+var ocpPayloadTestsPattern = regexp.MustCompile(`(?mi)^/ocp-payload-tests\s*$`)
+
+func helpProvider(_ []prowconfig.OrgRepo) (*pluginhelp.PluginHelp, error) {
+	// TODO(muller): Better descriptions, better help
+	pluginHelp := &pluginhelp.PluginHelp{
+		Description: `The ocp-payload-testing plugin spawns a battery of OCP release qualification testsuite jobs`,
+	}
+	pluginHelp.AddCommand(pluginhelp.Command{
+		// TODO(muller): What commands do we actually need?
+		// We will likely want to specify which jobs: blocking | informing | analysis | periodics | all
+		// We will also likely want to specify which release: ci | nightly | arm64 | s390x | ppc64
+		Usage:       "/ocp-payload-tests",
+		Description: "The ocp-payload-testing plugin spawns a battery of OCP release qualification testsuite jobs",
+		WhoCanUse:   "Members of the trusted organization for the repo.",
+		Examples:    []string{"/ocp-payload-tests blocking"},
+	})
+	return pluginHelp, nil
+}
+
+type server struct {
+	ghc                githubClient
+	ciopResolverClient configresolver.ResolverClient
+}
+
+func (s *server) handleIssueComment(l *logrus.Entry, ic github.IssueCommentEvent) {
+	if ic.Issue.IsPullRequest() {
+		return
+	}
+
+	if !ocpPayloadTestsPattern.MatchString(ic.Comment.Body) {
+		return
+	}
+
+	meta := api.Metadata{
+		Org:    ic.Repo.Owner.Login,
+		Repo:   ic.Repo.Name,
+		Branch: "", // TODO(muller): How to get branch from Issue
```

Comment:
I think you have to

```go
pr, err := ghc.GetPullRequest(ic.Repo.Owner.Login, ic.Repo.Name, ic.Issue.Number)
```
And then you should have the branch info from `pr.Bare.Ref`